src/lib/tableland-calls.ts

```diff
@@ -36,44 +53,32 @@ async function parseResponse(res: any): Promise<any> {
   return json;
 }
 
-// Take an Object with any symantic for key naming and return a new Object with keys that are lowerCamelCase
-// Example: `camelCaseKeys({structure_hash: "123"})` returns `{structureHash: "123"}`
-function camelCaseKeys(obj: any): any {
-  return Object.fromEntries(
-    Object.entries(obj).map((entry: KeyVal) => {
-      const key = entry[0];
-      const val = entry[1];
-      return [camelCase(key), val];
-    })
-  );
-}
-
 async function GeneralizedRPC(
   this: Connection,
   method: string,
   params: RpcParams = {}
 ) {
-  const signer = this.signer;
-  const address = await signer.getAddress();
-
-  const param: RpcParams = { controller: address, ...params };
-
   return {
     jsonrpc: "2.0",
     method: `tableland_${method}`,
     id: 1,
-    params: [param],
+    params: [params],
   };
 }
 
 async function hash(
   this: Connection,
   query: string
-): Promise<StructureHashReceipt> {
+): Promise<StructureHashResult> {
   const message = await GeneralizedRPC.call(this, "validateCreateTable", {
     create_statement: query,
   });
-  const json = await SendCall.call(this, message);
+  if (!this.token) {
```

Comment:
lazily fetch token, and lazily fetch signer, and only cache them if we update them